src/lib/Scenes/Artwork/Components/RequestConditionReport.tsx

```diff
@@ -0,0 +1,186 @@
+import { RequestConditionReport_artwork } from "__generated__/RequestConditionReport_artwork.graphql"
+import { RequestConditionReport_me } from "__generated__/RequestConditionReport_me.graphql"
+import { RequestConditionReportMutation } from "__generated__/RequestConditionReportMutation.graphql"
+import { RequestConditionReportQuery } from "__generated__/RequestConditionReportQuery.graphql"
+import { Modal } from "lib/Components/Modal"
+import { defaultEnvironment } from "lib/relay/createEnvironment"
+import { Schema, track } from "lib/utils/track"
+import React from "react"
+import { View } from "react-native"
+import { commitMutation, createFragmentContainer, graphql, QueryRenderer } from "react-relay"
+import { PayloadError } from "relay-runtime"
+
+interface RequestConditionReportProps {
+  artwork: RequestConditionReport_artwork
+  me: RequestConditionReport_me
+}
+
+interface State {
+  requestingConditionReport: boolean
+  showConditionReportRequestedModal: boolean
+  showErrorModal: boolean
+  errorModalText: string
+}
+
+@track({
+  action_name: Schema.ActionNames.RequestConditionReport,
+  context_module: Schema.ContextModules.ArtworkDetails,
+})
+export class RequestConditionReport extends React.Component<RequestConditionReportProps, State> {
+  state = {
+    requestingConditionReport: false,
+    showConditionReportRequestedModal: false,
+    showErrorModal: false,
+    errorModalText: "",
+  }
+
+  requestConditionReport = () => {
+    const { artwork } = this.props
+    return new Promise(async (resolve, reject) => {
+      commitMutation<RequestConditionReportMutation>(defaultEnvironment, {
+        onCompleted: data => {
+          resolve(data)
+        },
+        onError: error => {
+          reject(error)
+        },
+        mutation: graphql`
+          mutation RequestConditionReportMutation($input: RequestConditionReportInput!) {
+            requestConditionReport(input: $input) {
+              conditionReportRequest {
+                internalID
+              }
+            }
+          }
+        `,
+        variables: {
+          input: { saleArtworkID: artwork.saleArtwork.internalID },
+        },
+      })
+    })
+  }
+
+  @track({
+    action_type: Schema.ActionTypes.Fail,
+  })
+  presentErrorModal(errors: Error | ReadonlyArray<PayloadError>, mutationMessage: string) {
+    console.error("RequestConditionReport.tsx", errors)
+    const errorMessage = mutationMessage || "There was a problem processing your request. Please try again."
+    this.setState({ showErrorModal: true, errorModalText: errorMessage, requestingConditionReport: false })
+  }
+
+  @track({
+    action_type: Schema.ActionTypes.Success,
+  })
+  presentSuccessModal() {
+    this.setState({ showConditionReportRequestedModal: true, requestingConditionReport: false })
+  }
+
+  @track({
+    action_type: Schema.ActionTypes.Tap,
+  })
+  handleRequestConditionReportTap() {
+    this.setState({ requestingConditionReport: true })
```

Comment:
Not a big deal, but to be tidy I think you could manage this state value entirely inside the `requestConditionReport` method rather than in here and then later in the `present*Modal()` methods. You could set it to true at the beginning of `requestConditionReport `, and then set it to false in a `finally` clause on the promise, before the promise is returned.

Reply:
No you are right that is definitely cleaner, I don't think I can use finally unfortunately because it was introduced in [Ecmascript2018](https://www.ecma-international.org/ecma-262/9.0/index.html#sec-promise.prototype.finally) and we are using [2015](https://github.com/artsy/emission/blob/master/tsconfig.json#L7) I am wondering what consequences of upgrading are? That said I will consolidate to **catch** and **then** instead.

Reply:
ah didn't realise it was so new. `async` functions support `try {...} catch (...) {...} finally { ... }` but in this case it's probably not worth rewriting.